### ISSUE TITLE
Most symbols not found

### ISSUE THREAD

**Issue by jyavenard** [microsoft/vscode-cpptools#637]
Most symbols resolution no longer work following the upgrade to VSCode 1.11 and the latest Cpptools extension (currently 0.10.5)
STR:
Checkout mozilla-central repository:
hg clone https://hg.mozilla.org/mozilla-central/

Open VSCode
Open the directory mozilla-central (that you checked out)
Open the file WebrtcOMXH264VideoCodec.h (Command-P WebrtcOMXH264VideoCodec.h) found in ./media/webrtc/signaling/src/media-conduit/WebrtcOMXH264VideoCodec.h

You can see that that the #include "AudioConduit.h" has a green squiggle, so click on it and add the path where it's found:
"${workspaceRoot}/media/webrtc/signaling/src/media-conduit",
(I tried with the absolute path too)

This will kick the Microsoft.VCCode.CPP.Extension.Darwin (I'm on mac) process at 100% for several minutes. I'm guessing it's rescanning the code refreshing the database.

#include "AudioConduit.h" will always have the green squiggle attached to it.

Now open say VideoConduit.cpp (same directory)
go to line 279, on function:
bool WebrtcVideoConduit::SetLocalSSRCs(const std::vector<unsigned int> & aSSRCs)

Locate the line mSendStreamConfig.rtp.ssrcs = aSSRCs;
Right click on the mSendStreamConfig name, select "Go To Definition" it will show the message "No definition found for 'mSendStreamConfig'"

Yet, hover the mouse on the line above, above the word Update in the comment:
// Update the value of the ssrcs in the config structure.

And here it gives me plenty of views of what "Update" is (surely it should ignore comments)

Mind you those are just examples, almost no symbols can ever be found now.
It used to work very well :(

As it is, VSCode has become unusable with our C++ project.

Thanks for looking into it.

**Comment by sean-mcmanus** [comment]
The bug doesn't repro for me. The AudioConduit.h is in the same directory as the *VideoCodec.h so you shouldn't need any special includePath. We don't show comments on hover, so it sounds like you're using another extension, which could be interfering with our Go To Definition. Can you give your extension list or disable the other extensions? Or maybe I'm misunderstanding your comment. You can try deleting the browse.vc.db or changing the databaseFilename to rebuild the symbol database.

**Comment by jyavenard** [comment]
I know you shouldn't have to enter the extra path. It was just to show that the header wasn't found and even when manually adding the path it is still not found. 

This is happening on mac.

**Comment by jyavenard** [comment]
I did delete all vscode related folders in ~/Library, including the db file in order to make sure the files were rescanned. 
the cpp confit is the default one automatically generated. 

Will post the list of extensions in my next post.

**Comment by sean-mcmanus** [comment]
Yes, I'm using Mac too. It isn't reproing for me. Is the Mac in the bottom right red still? If any symbolic links were added to the workspace which point to somewhere else in the workspace that could cause a problem.

**Comment by jyavenard** [comment]
- VSCode Version: Code 1.11.2 (6eaebe3b9c70406d67c97779468c324a7a95db0e, 2017-04-13T07:56:42.517Z)
- OS Version: Darwin x64 16.5.0
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|autotrim|NathanRidley|1.0.6|
|code-gnu-global|austin|0.2.1|
|git-easy|bibhasdn|1.9.1|
|githistory|donjayamanne|0.2.0|
|CppSnippets|hars|0.0.7|
|vscode-rust|kalitaalexey|0.3.11|
|vscode-clang|mitaki28|0.2.2|
|cpptools|ms-vscode|0.10.5|
|annotator|ryu1kn|0.10.1|
|rewrap|stkb|1.3.0|
|vscode-lldb|vadimcn|0.4.0|
|gitblame|waderyan|1.3.0|;


Building the code does create an obj directory, that contains all the headers as symbolic link elsewhere.. However, after reading that the new version of cpptools had trouble with that, I made the object directory be located one level up from the current directory

**Comment by jyavenard** [comment]
<img width="830" alt="image" src="https://cloud.githubusercontent.com/assets/505748/25055244/e9a0100e-2161-11e7-8e0b-4c1652ca7506.png">

**Comment by jyavenard** [comment]
After recreating the database:
`{
    "configurations": [
        {
            "name": "Mac",
            "includePath": [
                "/usr/include",
                "/usr/local/include"
            ],
            "browse": {
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": "/Users/jyavenard/Work/Mozilla/vscode.db"
            }
        },
        {
            "name": "Linux",
            "includePath": [
                "/usr/include",
                "/usr/local/include"
            ],
            "browse": {
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
        },
        {
            "name": "Win32",
            "includePath": [
                "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include/*"
            ],
            "browse": {
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
        }
    ]
}
`
So now I have (Code Helper having completed)
$ ls -l ../vscode.db
-rw-r--r--  1 jyavenard  staff  1486848 14 Apr 22:34 ../vscode.db

I no longer see a green squiggle next to the header (I also deleted code-gnu-global and vscode-clang)
However:

symbol search is still barely working
![image](https://cloud.githubusercontent.com/assets/505748/25055446/1f442366-2163-11e7-9824-c4ecd75a4f3d.png)

**Comment by jyavenard** [comment]
Side note, and likely related to issue #612; I can switch from .cpp to .h file by pressing Alt-O; but the .h -> .cpp does absolutely nothing

**Comment by sean-mcmanus** [comment]
I can repro the bug with SetSeekThreshold, but not the one with WebrtcVideoEncoder. It looks like VideoConduit.h is failing to get parsed for some reason. If building that project causes folders to get created, it's possible those files are causing our parser to crash (adding the build output to files.exclude might fix the issue). Do you see anything recent at `Users/…/Library/Logs/DiagnosticReports/<process><date><machinename>.crash`

**Comment by jyavenard** [comment]
Nothing over the past few days:
```
$ ls -l ~/Library/Logs/DiagnosticReports/Microsoft.VSCode.CPP.Extension.darwin_2017-04-*
-rw-------@ 1 jyavenard  staff  31599  8 Apr 10:14 /Users/jyavenard/Library/Logs/DiagnosticReports/Microsoft.VSCode.CPP.Extension.darwin_2017-04-08-101403_jyapro.crash
-rw-------@ 1 jyavenard  staff  31998  8 Apr 15:49 /Users/jyavenard/Library/Logs/DiagnosticReports/Microsoft.VSCode.CPP.Extension.darwin_2017-04-08-154908_jyapro.crash
-rw-------@ 1 jyavenard  staff  31997 10 Apr 10:00 /Users/jyavenard/Library/Logs/DiagnosticReports/Microsoft.VSCode.CPP.Extension.darwin_2017-04-10-100045_jyapro.crash

```

**Comment by jyavenard** [comment]
> adding the build output to files.exclude might fix the issue

side question: Is it possible to exclude the parent object directory, yet include some sub-directories in that? Some headers are generated at compile time (ipdl files)

At this stage, that can't be it, the build directory isn't in the workspace directory. (it's set to be ../${workspaceRoot}/obj-dbg)

BTW, I've noticed that relative folders do not work c_cpp_properties.json config..

**Comment by sean-mcmanus** [comment]
What do you mean relative folders don't work? ${workspaceRoot}/../subDir is supposed to work. You should be able to use that to include a parent folder subdirectory but not the parent folder itself.

**Comment by jyavenard** [comment]
It's now been a month where VS Code have just been completely unusable for me on mac (the same folder on linux or windows work just fine)

Alt-O to switch between headers and source code never works.
No symbols of any kind can be resolved.

I have fully uninstalled VS Code, all its preferences, indexes, and so forth... no better...

What can I do to help solve this issue?

**Comment by jyavenard** [comment]
Here is my c_cpp_properties.json (default as generated by VS code)

```
{
    "configurations": [
        {
            "name": "Mac",
            "includePath": [
                "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1",
                "/usr/local/include",
                "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/8.1.0/include",
                "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include",
                "/usr/include",
                "/usr/include/machine",
                "${workspaceRoot}"
            ],
            "defines": [],
            "intelliSenseMode": "clang-x64",
            "browse": {
                "path": [
                    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1",
                    "/usr/local/include",
                    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/8.1.0/include",
                    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include",
                    "/usr/include",
                    "/usr/include/machine",
                    "${workspaceRoot}"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
        },
        {
            "name": "Linux",
            "includePath": [
                "/usr/include",
                "/usr/local/include",
                "${workspaceRoot}"
            ],
            "defines": [],
            "intelliSenseMode": "clang-x64",
            "browse": {
                "path": [
                    "/usr/include",
                    "/usr/local/include",
                    "${workspaceRoot}"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
        },
        {
            "name": "Win32",
            "includePath": [
                "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include",
                "${workspaceRoot}"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE"
            ],
            "intelliSenseMode": "msvc-x64",
            "browse": {
                "path": [
                    "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include/*",
                    "${workspaceRoot}"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
        }
    ],
    "version": 2
}
```

I have once again deleted the database, selected C/Cpp: Reset Intellisense Database
to no available :(

**Comment by jyavenard** [comment]
Doing Cmd-Shift-O to show the list of symbols, shows 92 symbols only. Selecting any of them jump to a place that has ZERO relationship with the symbol name.
Sometimes in the middle of a function definition or within a comment.

**Comment by jyavenard** [comment]
My source code was located in the directory ~/Work/Mozilla/mozilla-central

After doing the following.
rename:
$ ~/Work/Mozilla/mozilla-central to ~/Work/Mozilla/mozilla-central.bak
$ mkdir central
$ rsync -avP mozilla-central.bak/.git/ central/.git/
$ cd central
$ git reset --hard HEAD
$ re-open VS Studio, point to the ~/Work/Mozilla/central directory

Now things are working as they used to.

Very weird.

**Comment by jyavenard** [comment]
this now works for me...
puzzled on what happened. but anyhow..

**Comment by jyavenard** [comment]
Now I can reproduce on Windows too...

**Comment by sean-mcmanus** [comment]
@jyavenard We fixed some bugs with 0.14.6 that may have caused this. Are you still experiencing problems?

**Comment by bobbrow** [comment]
I'm closing old issues marked "more info needed".  If you are still having issues with the extension, please ensure you have installed the latest version and open a new issue with the issue details.
